extensions/metadata_exchange/plugin.cc

```diff
@@ -105,15 +105,16 @@ bool PluginRootContext::onConfigure(size_t size) {
   const std::string function = "declare_property";
   envoy::source::extensions::common::wasm::DeclarePropertyArguments args;
   args.set_type(envoy::source::extensions::common::wasm::WasmType::FlatBuffers);
-  args.set_span(
-      envoy::source::extensions::common::wasm::LifeSpan::DownstreamConnection);
+  args.set_span(envoy::source::extensions::common::wasm::LifeSpan::FilterChain);
```

Comment:
should we have a test to verify the appropriate behavior?  This feels like something we should have validation on.

Reply:
Absolutely, and also discussion about lifetimes.
@PiotrSikora @jplevyak @lambdai can you take a look at this issue?

Reply:
Damn, my hunch is that FilterChain has longer life than connection until I read the document. It seems FilterChain here means the http filters in HCM.